reco_utils/dataset/criteo_dac.py

```diff
@@ -1,73 +1,164 @@
+    More details: http://labs.criteo.com/2013/12/download-terabyte-click-logs/
 
-def _load_datafile(local_cache_path="dac_sample.tar.gz"):
-    """ Download and extract file """
+    Args:
+        size (str): Dataset size. It can be "sample" or "full".
+        local_cache_path (str): Path where to cache the tar.gz file locally
+        header (list): Dataset header names.
+
+    Returns:
+        pd.DataFrame: Criteo DAC sample dataset.
+    """
+    path, filename = os.path.split(os.path.realpath(local_cache_path))
+    download_criteo(size, filename, path)
+    filepath = extract_criteo(size, path, local_cache_path)
+    return pd.read_csv(filepath, sep="\t", header=None, names=header)
+
+
+def load_spark_df(
+    spark,
+    size="sample",
+    header=DEFAULT_HEADER,
+    local_cache_path="dac.tar.gz",
+    dbfs_datapath="dbfs:/FileStore/dac",
+    dbutils=None,
+):
+    """Loads the Criteo DAC dataset as pySpark.DataFrame.
+
+    The schema is:
+    <label> <integer feature 1> ... <integer feature 13> <categorical feature 1> ... <categorical feature 26>
+
+    More details: http://labs.criteo.com/2013/12/download-terabyte-click-logs/
+
+    Args:
+        spark (pySpark.SparkSession): Spark session.
+        size (str): Dataset size. It can be "sample" or "full".
+        local_cache_path (str): Path where to cache the tar.gz file locally.
+        header (list): Dataset header names.
+        dbfs_datapath (str): Where to store the extracted files on Databricks.
+        dbutils (Databricks.dbutils): Databricks utility object.
+  
+    Returns:
+        pySpark.DataFrame: Criteo DAC training dataset.
+    """
     path, filename = os.path.split(os.path.realpath(local_cache_path))
-    # Make sure a temporal zip file get cleaned up no matter what
-    atexit.register(_clean_up, local_cache_path)
+    download_criteo(size, filename, path)
+    filepath = extract_criteo(size, path, local_cache_path)
+
+    if is_databricks():
+        try:
+            # Driver node's file path
+            tar_datapath = "file:" + filepath
+            ## needs to be on dbfs to load
+            dbutils.fs.cp(tar_datapath, dbfs_datapath, recurse=True)
+            path = dbfs_datapath
+        except:
+            raise ValueError(
+                "To use on a Databricks notebook, dbutils object should be passed as an argument"
+            )
+    else:
+        path = filepath
+
+    schema = _get_spark_schema(header)
+    return spark.read.csv(path, schema=schema, sep="\t", header=False)
+
+
+def download_criteo(size="full", filename="dac.tar.gz", work_directory="."):
+    """Download criteo dataset as a compressed file.
+
+    Args:
+        size (str): Size of criteo dataset. It can be "full" or "sample".
+        filename (str): Filename.
+        work_directory (str): Working directory.
+
+    Returns:
+        str: Path of the downloaded file.
+
+    """
+    url = CRITEO_URL[size]
+    return maybe_download(url, filename, work_directory)
```

Comment:
how is cleanup of the extracted files handled? We shouldn't surface that outside the data loader

Reply:
changes based on @loomlike PR